### ISSUE TITLE
Provide a convenient way to convert to Surface's pixel format

### ISSUE THREAD

**Issue by ilyapopov** [libSDL2pp/libSDL2pp#61]
Suppose I want to fill a rectangle with some color. `FillRect` member function expects a color in surface's format. There is conversion function, [`SDL_MapRGB`](https://wiki.libsdl.org/SDL_MapRGB), but it expects [`SDL_PixelFormat*`](https://wiki.libsdl.org/SDL_PixelFormat). `Surface::GetFormat`, however, returns only `SDL_PixelFormat::format`, not the whole structure.

What is the preferred way to convert RGB values to surface's color that `FillRect` expects?

```
surface.FillRect(rect, SDL_MapRGB(???, r, g, b));
```

**Comment by ooxi** [comment]
:+1:

**Comment by AMDmi3** [comment]
Yes, I've had that in plans. Not yet figured out a consistent way to handle both integer pixelformat and SDL_PixelFormat structure everywhere (renderer, texture, surface). Regarding the color, MapRGB should likely be a method of a SDL_PixelFormat wrapper. I'll try to handle this next year :)

**Comment by ilyapopov** [comment]
Is this a correct workaround?

```
SDL_PixelFormat fmt* SDL_AllocFormat(surface.Format());

surface.FillRect(rect, SDL_MapRGB(fmt, r, g, b));

SDL_FreeFormat(fmt);

**Comment by ilyapopov** [comment]
Or better this:

```
std::unique_ptr<SDL_PixelFormat, void (*)(SDL_PixelFormat *)> fmt(
    SDL_AllocFormat(surface.Format()), SDL_FreeFormat);
surface.FillRect(rect, SDL_MapRGB(fmt.get(), r, g, b));

**Comment by AMDmi3** [comment]
You don't have to allocate a new pixel format as Surface already contains one.

Probably something like this:
```
surface.FillRect(rect, SDL_MapRGB(surface.Get()->format, r, g, b));
```

**Comment by ilyapopov** [comment]
Thanks! I did not realize i can access `format` through `Get()`.
